lib/spree_multi_domain/multi_domain_helpers.rb

```diff
@@ -25,7 +25,7 @@ def add_current_store_id_to_params
 
     private
     def store_key
-      request.headers['HTTP_SPREE_STORE'] || request.env['SERVER_NAME']
+      current_store_class.new(request).send(:store_key)
```

Comment:
I'm a little concerned about calling a private method on another object here. I understand that the reason you did that was because `Spree::Tracker.current` accepts a `store_key` rather than a `store_id`, in which case we could've just called `current_store_class.new(request).store` and called it a day. 

I appreciate the removal of duplicate functionality though.